src/core/hle/service/nvflinger/buffer_queue.cpp

```diff
@@ -115,11 +115,36 @@ std::optional<std::reference_wrapper<const BufferQueue::Buffer>> BufferQueue::Ac
     return *itr;
 }
 
-void BufferQueue::ReleaseBuffer(u32 slot) {
+std::optional<std::reference_wrapper<const BufferQueue::Buffer>>
+BufferQueue::ObtainPresentBuffer() {
+    auto itr = queue.end();
+    // Iterate to find a queued buffer matching the requested slot.
+    while (itr == queue.end() && !presenting_sequence.empty()) {
+        u32 slot = presenting_sequence.front();
+        itr = std::find_if(queue.begin(), queue.end(), [&slot](const Buffer& buffer) {
```

Comment:
```suggestion
        itr = std::find_if(queue.begin(), queue.end(), [slot](const Buffer& buffer) {
```

This is just a primitive type that's smaller than a pointer, there's no need to take it by reference.